competition-simulator.md

```diff
@@ -0,0 +1,131 @@
+
+```
+.
+├── competition-simulator
+│   ├── ...
+│   └─ worlds
+│       └── Arena.wbt
+└── robot.py
+```
+
+Your code should be developed in `robot.py`.
+
+
+<div class="warning">
+  Only your controller code will be present in the competition environment.
+</div>
+
+<div class="warning">
+  You will be responsible for backing up and versioning your code, and collaborating with your fellow team members. The <a href="https://studentrobotics.org/ide/">IDE</a> is unsuitable for this simulator.
+</div>
+
+### Updates
+
+Occasionally, we may release an update to the simulation. To update, you will need to delete the `competition-simulator-master` directory, and re-download it using the above link.
+
+## Robot
+
+![]({{ site.baseurl }}/resources/competition-simulator/robot-front.png)
+
+## Programming Interface
+
+Unless otherwise stated, the simulator’s API is the same as the real SR API described in the [programming docs]({{ site.baseurl }}/programming/).
+
+### Motors
+
+Your robot has 2 motor boards attached, each with 2 motors. Board `0` has the left wheel in port `m0`, and the right wheel in `m1`. Board `1` has the gripper lift motor in `m0`, and the finger motors in `m1`.
+
+### Ruggeduino
+
+Your robot has 5 microswitches and 6 distance sensors, attached to the digital and analogue pins respectively. These are all attached to a single ruggeduino.
+
+Because these sensors are pre-attached to the ruggeduino, you do not need to set its `pin_mode`.
+
+#### Microswitches
+
+The microswitches are attached to digital pins 2-6:
+
+- Front
+- Back
+- Between gripper fingers
+- Left gripper finger
+- Right gripper finger
+
+These are shown as red coloured blocks on the robot. Using the `digital_read`  method, you'll receive a `bool` telling you whether the switch is current actuated.
+
+#### Distance Sensors
+
+Analogous to ultrasound sensors, distance sensors allow you to retrieve the distance between your robot and an object. These are attached to analogue pins 0-5:
+
+- Front Left
+- Front Right
+- Left
+- Right
+- Back Left
+- Back Right
+
+These are shown as blue coloured blocks on the robot. The `analogue_read` method will return the distance in metres, however only measure up to 30cm.
+
+### Camera
+
+Your robot has a camera, which is attached to the top of your robot. A live preview of what the camera sees is shown in the top-left corner. The vision system uses Webots' object recognition, rather than [fiducial markers]({{ site.baseurl }}/programming/sr/vision/markers/).
+
+The `see` method will return a list of visible markers in the arena, but doesn't allow a resolution argument. Each token is as described in the [vision docs]({{ site.baseurl }}/programming/sr/vision/), except:
+
+- `Marker.info.marker_type` will be one of `MarkerType.SILVER`, `MarkerType.GOLD` or `MarkerType.ARENA`. `MarkerType` can be imported with `from sr.robot import MarkerType`
```

Comment:
Should we also expose the constants named in the docs?

Reply:
I feel like it should be 1 or the other. Either it should be compatible, or it shouldn't.

Personally I like the idea of using enums.

Reply:
I would err on the side of compatibility here. I agree that enums are nice, however I think the closer we can make the code to their existing code the better.

(I'd be tempted to leave the `MarkerType` enum exported anyway, so teams can use it if they want to, just to also add the named constants from the API docs. I realise this violates the Zen of Python, but I think the downsides are small in this case)

Reply:
https://github.com/srobo/competition-simulator/pull/85

Reply:
Ah, sorry, I was thinking that we wouldn't necessarily document the enum. I don't think it matters a great deal now it's done though.